pandas/errors/__init__.py

```diff
@@ -38,9 +38,53 @@ class ParserError(ValueError):
 
 class DtypeWarning(Warning):
     """
-    Warning that is raised for a dtype incompatibility. This
-    can happen whenever `pd.read_csv` encounters non-
-    uniform dtypes in a column(s) of a given CSV file.
+    Warning raised when importing different dtypes in a column from a file.
+
+    Raised for a dtype incompatibility. This can happen whenever `pd.read_csv`
+    or `pd.read_table` encounter non-uniform dtypes in a column(s) of a given
+    CSV file.
+
+    It only happens when dealing with larger files.
+
+    See Also
+    --------
+    pd.read_csv : Read CSV (comma-separated) file into a DataFrame.
+    pd.read_table : Read general delimited file into a DataFrame.
+
+    Notes
+    -----
+    Despite the warning, the CSV file is imported with mixed types in a single
+    column. See the examples below to better understand this issue.
+
+    Examples
+    --------
+    This example creates and reads a large CSV file with a column that contains
+    `int` and `str`.
+
+    >>> df = pd.DataFrame({'a':['1']*100000 + ['X']*100000 + ['1']*100000,
+    ...                    'b':['b']*300000})
+    >>> df.to_csv('test', sep='\t', index=False, na_rep='NA')
+    >>> df2 = pd.read_csv('test', sep='\t')
+    Traceback (most recent call last):
```

Comment:
I think if yyou add a line like

    >>> df2 = pd.read_csv('test', sep='\t')
    ... # doctest: +SKIP

The doctest might work.

Reply:
Yeah, I don't think it will be possible to get Warnings to work with doctest, so adding a `# doctest: +SKIP` is fine

Reply:
I'm sorry but I couldn't understand what the doctest: +SKIP is supposed to do. I've added it right after the read_csv and i just got an error. Can you help me?

################################################################################
################################### Doctests ###################################
################################################################################

**********************************************************************
Line 28, in pandas.errors.DtypeWarning
Failed example:
    df2 = pd.read_csv('test', sep=' ')
Expected:
    # doctest: +SKIP
    DtypeWarning: Columns (0) have mixed types...
Got nothing